docs/editing-components.md

```diff
@@ -0,0 +1,408 @@
+    for more details.
+
+
+### The front end
+
+#### The HTML template
+
+Frontend rendering of Wagtail StreamField blocks to HTML
+can be controlled by writing a Django template
+and associating it with the block type.
+A custom block definition can specify the template path in its `Meta` class:
+
+```python
+from wagtail.core import blocks
+
+
+class PersonBlock(blocks.StructBlock):
+    name = blocks.CharBlock()
+    email = blocks.EmailBlock()
+
+    class Meta:
+        template = 'myapp/blocks/person_block.html'
+```
+
+StreamField block templates are loaded by the Django template loader
+in the same way that Django page templates are.
+The specified template path must be loadable
+by one of the Django template engines configured in
+[`settings.TEMPLATES`](https://docs.djangoproject.com/en/1.11/ref/settings/#std:setting-TEMPLATES).
+(This project supports both the standard
+[Django templates backend](https://docs.djangoproject.com/en/2.1/topics/templates/#django.template.backends.django.DjangoTemplates)
+and the
+[Jinja2 backend](https://docs.djangoproject.com/en/2.1/topics/templates/#django.template.backends.jinja2.Jinja2),
+but Jinja2 is more commonly used.)
+[See the Django templates documentation](https://docs.djangoproject.com/en/1.11/topics/templates/#usage)
+for more details on the search algorithm used to locate a template.
+
+Returning to the `RelatedContent` example,
+this is what its Jinja2 template looks like (comments excluded):
+
+```html
+<div class="m-related-content">
+    {% if value.heading %}
+        <header class="m-slug-header">
+            <h2 class="a-heading">
+                {{ value.heading }}
+            </h2>
+        </header>
+    {% endif %}
+
+    {{ value.paragraph | safe }}
+
+    {% if value.links %}
+        <ul class="m-list m-list__links">
+        {% for link in value.links %}
+            <li class="m-list_item">
+                <a href="{{ link.url }}" class="m-list_link">{{ link.text }}</a>
+            </li>
+        {% endfor %}
+        </ul>
+    {% endif %}
+</div>
+```
+
+When Wagtail renders a block,
+it includes the values of its fields in an object named `value`.
+Above, you can see where the `heading` and `paragraph` fields are output with
+[Jinja2 expression tags](http://jinja.pocoo.org/docs/2.10/templates/#expressions).
+And note how the `links` field (a `ListBlock`) is iterated over,
+and the values of _its_ `Hyperlink` child blocks are output.
+
+That's about as simple an example as it gets, but block templates can get much
+more complex when they have lots of child blocks and grandchild blocks.
+Also, if a block definition has overridden `get_context` to pass other data
+into the template (as described at the end of
+[the Python class section](#the-python-class) above),
+those context variables can also be output with simple Jinja2 expression tags:
+`{{ context_var }}`.
+
+#### Adding some style
```

Comment:
I would prefer this heading be more straightforward like "Adding CSS" or just "CSS" so that a quick scan of the TOC or the page is more likely to grab a reader's eye

Reply:
Addressed in 89f23454e89e7befda7ec3cbe69e2b4ca61cd59e